frontend/src/components/search/ContactResultsTable.js

```diff
@@ -0,0 +1,121 @@
+  query: PropTypes.string.isRequired,
+  partyRelationshipTypeHash: PropTypes.objectOf(PropTypes.strings).isRequired,
+  showAdvancedLookup: PropTypes.bool.isRequired,
+};
+
+const defaultProps = {};
+
+const parseQuery = (query) => {
+  const emailExp = /@/;
+  const phoneExp = /^\d+(-\d*)*$/;
+  if (emailExp.test(query)) {
+    return { email: query };
+  }
+  if (phoneExp.test(query)) {
+    return { phone_no: query };
+  }
+  const terms = query.split(" ");
+  if (terms.length === 1) {
+    return { last_name: terms[0] };
+  }
+  return { first_name: terms[0], last_name: terms.slice(1).join(" ") };
+};
+
+export const ContactResultsTable = (props) => {
+  const columns = [
+    {
+      title: "Party Guid",
+      dataIndex: "party_guid",
+      key: "party_guid",
+      render: (text, record) => [
+        <Row>
+          <Col span={24}>
+            <Link
+              to={router.PARTY_PROFILE.dynamicRoute(record.party_guid)}
+              style={{ fontSize: "1.25rem" }}
+            >
+              <Highlight search={props.highlightRegex}>{record.name}</Highlight>
+            </Link>
+          </Col>
+        </Row>,
+        <Row className="padding-small--top">
+          <Col xs={24} md={8} lg={4}>
+            <p>Roles</p>
+          </Col>
+          <Col xs={24} md={16} lg={8}>
+            {props.partyRelationshipTypeHash.PMT &&
+              record.mine_party_appt.map((pr) => (
+                <p>
+                  {props.partyRelationshipTypeHash[pr.mine_party_appt_type_code]}
+                  <span className="padding-small--left" style={{ fontStyle: "italic" }}>
+                    ({pr.mine.mine_name})
+                  </span>
+                </p>
+              ))}
+          </Col>
+          <Col xs={24} md={8} lg={6} xxl={4}>
+            <p>Email</p>
+          </Col>
+          <Col xs={24} md={16} lg={12} xxl={8}>
+            <p>
+              <Highlight search={props.highlightRegex}>{record.email}</Highlight>
+            </p>
+          </Col>
+          <Col xs={24} md={8} lg={6} xxl={4}>
+            <p>Phone</p>
+          </Col>
+          <Col xs={24} md={16} lg={12} xxl={8}>
+            <p>
+              <Highlight search={props.highlightRegex}>{record.phone_no}</Highlight>
+              {record.phone_ext && record.phone_ext}
+            </p>
+          </Col>
+        </Row>,
+      ],
+    },
+  ];
+  return (
+    <div>
+      <h2>{props.header}</h2>
+      <Divider style={{ padding: "0" }} />
+      <Table
+        className="nested-table padding-large--bottom"
+        align="left"
+        showHeader={false}
+        pagination={false}
+        columns={columns}
+        dataSource={props.searchResults}
+      />
+      {props.showAdvancedLookup && (
+        <Link
+          style={{ float: "right", fontSize: "1.25rem" }}
+          className="padding-large--left"
+          to={router.CONTACT_HOME_PAGE.dynamicRoute(parseQuery(props.query))}
```

Comment:
`router.CONTACT_HOME_PAGE.dynamicRoute` expects `page` and `per_page` params, I don't see these being set anywhere? Does this behave as expected?

Reply:
It does work, although I notice the page count looks wrong. The contact and mine page should probably be responsible for setting its default parameters if hit. I'll see if that's quick to do.

Reply:
Added page and per_page params!